documents/attribute.go

```diff
@@ -164,20 +188,56 @@ type Attribute struct {
 
 // NewAttribute creates a new custom attribute.
 func NewAttribute(keyLabel string, attrType AttributeType, value string) (attr Attribute, err error) {
-	if keyLabel == "" {
-		return attr, ErrEmptyAttrLabel
+	attrKey, err := AttrKeyFromLabel(keyLabel)
+	if err != nil {
+		return attr, err
+	}
+
+	attrVal, err := AttrValFromString(attrType, value)
+	if err != nil {
+		return attr, err
 	}
 
+	return Attribute{
+		KeyLabel: keyLabel,
+		Key:      attrKey,
+		Value:    attrVal,
+	}, nil
+}
+
+// NewSignedAttribute returns a new signed attribute
+// takes keyLabel, signer identity, signer account, model and value
+// signature payload: sign(identity + docID + docVersion + value + time)
+func NewSignedAttribute(keyLabel string, identity identity.DID, account config.Account, model Model, value []byte) (attr Attribute, err error) {
 	attrKey, err := AttrKeyFromLabel(keyLabel)
 	if err != nil {
 		return attr, err
 	}
 
-	attrVal, err := AttrValFromString(attrType, value)
+	// TODO(ved): time should be part of the signature
+	var signPayload []byte
```

Comment:
Can you create a helper method that takes a Signed field struct and constructs the payload from it and then reuse it below?